packages/core/src/tools/tool-builder/builder.ts

```diff
@@ -209,36 +158,30 @@ export class CoreToolBuilder extends MastraBase {
         : undefined,
     };
 
-    const parametersObject: { parameters?: ZodType; inputSchema?: ZodType } = {};
-
-    if (isVercelTool(this.originalTool)) {
-      parametersObject.parameters = this.getParameters();
-    } else {
-      parametersObject.inputSchema = this.getParameters();
-    }
-
     const model = this.options.model;
 
-    const hasParameters = parametersObject.parameters || parametersObject.inputSchema;
-
-    if (model && hasParameters) {
-      for (const compat of [
-        new OpenAIReasoningToolCompat(model),
-        new OpenAIToolCompat(model),
-        new GoogleToolCompat(model),
-        new AnthropicToolCompat(model),
-        new DeepSeekToolCompat(model),
-        new MetaToolCompat(model),
-      ]) {
-        if (compat.shouldApply()) {
-          return { ...definition, ...compat.process({ ...this.originalTool, ...parametersObject }) };
-        }
-      }
+    const schemaCompatLayers = [];
+
+    if (model) {
+      schemaCompatLayers.push(
+        new OpenAIReasoningSchemaCompatLayer(model),
+        new OpenAISchemaCompatLayer(model),
+        new GoogleSchemaCompatLayer(model),
+        new AnthropicSchemaCompatLayer(model),
+        new DeepSeekSchemaCompatLayer(model),
+        new MetaSchemaCompatLayer(model),
+      );
```

Comment:
I do think we should provide a default way to apply all but it doesn't need to be in this PR.

Something like:
```ts
const processedSchema = applyCompatLayer({
       schema: this.getParameters(),
       compatLayers: 'all',
       mode: 'aiSdkSchema',
     });
```